src/database.py

```diff
@@ -0,0 +1,215 @@
+
+    # Aggregate order items to get total price and quantity
+    order_items_agg = order_items_with_products.group_by('order_id').agg([
+        pl.sum('price').alias('price'),
+        pl.sum('freight_value').alias('freight_value'),
+        pl.first('product_id'),
+        pl.first('product_category_name'),
+        pl.first('seller_id'),
+        pl.count('order_item_id').alias('item_quantity')
+    ])
+
+    # Aggregate payments to get total payments for each order
+    payments_agg = payments_df.group_by('order_id').agg([
+        pl.first('payment_installments'),
+        pl.sum('payment_value').alias('payment_value')
+    ])
+
+    # Get average review score by order_id
+    reviews_agg = reviews_df.group_by('order_id').agg([
+        pl.mean('review_score').cast(pl.Int32).alias('review_score')
+    ])
+
+    # Merge all DataFrames into a full orders DataFrame
+    orders_full = (
+        orders_with_customers
+        .join(order_items_agg, on='order_id', how='left')
+        .join(payments_agg, on='order_id', how='left')
+        .join(sellers_df, on='seller_id', how='left')
+        .join(reviews_agg, on='order_id', how='left')
+        .join(product_translation_df, on='product_category_name', how='left')
+    )
+
+    # Select necessary columns for the final DataFrame
+    final_orders_df = orders_full.select([
+        'order_id',
+        'customer_unique_id',
+        'order_status',
+        'order_purchase_timestamp',
+        'order_approved_at',
+        'order_delivered_carrier_date',
+        'order_delivered_customer_date',
+        'order_estimated_delivery_date',
+        'customer_city',
+        'price',
+        'freight_value',
+        'product_id',
+        'product_category_name',
+        'seller_id',
+        'item_quantity',
+        'payment_installments',
+        'payment_value',
+        'review_score',
+        'seller_city',
+        'product_category_name_english'
+    ])
+
+    logging.info("Orders DataFrame created.")
+    #print(final_orders_df.describe())  # Log the description of the DataFrame
+
+    return final_orders_df
+
+def insert_data_into_db(session, final_orders_df):
+    """Inserts processed data into the DuckDB database."""
+    logging.info("Inserting data into the database...")
+    
+    try:
+        records = final_orders_df.to_dicts()
+        for record in records:
+            order = Order(**record)
+            session.add(order)
+        session.commit()
+        logging.info("Data successfully inserted into the database.")
+
+    except Exception as e:
+        logging.error(f"An error occurred during data insertion: {e}")
+        session.rollback()
+
+    finally:
+        session.close()
+        logging.info("Database session closed.")
+
+def data_pipeline(WORKING_DIR):
+    ''' Runs the entire data pipeline'''
+    logging.info("Starting data pipeline...")
+    try:
+        # Set directory path
```

Comment:
you don't need this comments here like load data etc, comments are descriptions and should not be added when it is not needed, kindly apply this in other scripts

Reply:
Done